### PR TITLE
Remove dead code and update SLF4J dependency to 2.0.7

### LINKED ISSUE #750: Upgrading to SLF4J 2.x
I've noticed the following dependency in `pedestal.log`

https://github.com/pedestal/pedestal/blob/a51d99ea698dab20d46aff5ec837c8076079fee9/log/deps.edn#L15

What are the current blockers for upgrading to SLF4J 2.x? I'm interested in this version because it seems that Logback and a bunch of other libraries have moved on to the newer API, and I want to ensure that I can use versions of Logback with the latest security fixes.

### PR DESCRIPTION
**Pull request by hlship**
This also extends the CVE check to all dependencies across all modules.

Certain CVE surpressions were enabled, at least temporarily.

Tests executed by CI were modified to execute with both Clojure 1.10 and Clojure 1.11.

The *immutant* module was deleted outright.

Fixes #750

### CODE REVIEW

**Comment by hlship** [comment]
Closing to deal with more dependencies.

**Comment by hlship** [comment]
Tracking down why tests fails if the tests module adds a dependency to the service-tools module.

File: jetty/deps.edn [hunk 1]
```diff
@@ -23,12 +23,10 @@
...
         javax.servlet/javax.servlet-api {:mvn/version "3.1.0"}}
  :aliases
- {:clj11
-  {:override-deps {org.clojure/clojure {:mvn/version "1.11.1"}}}
-
+ {
   ;; clj -J-Dclojure.main.report=stderr -J-Dorg.slf4j.simpleLogger.log.org.apache.commons=error -X:nvd :classpath \""$(clojure -Alocal -Spath)\""
   :nvd
-  {:deps {nvd-clojure/nvd-clojure {:mvn/version "3.2.0"}}
+  {:deps {nvd-clojure/nvd-clojure {:mvn/version "3.3.0"}}
```
Comment (bonkydog) [inline jetty/deps.edn:29]: You mentioned that we could get rid of this too.

**Review by bonkydog — APPROVED** [review]
Great! :rocket:

**Pull request merged**